pkg/serializer/split/split.go

```diff
@@ -170,8 +171,15 @@ func serializeMarshaller(m marshaler.Marshaler, compress bool, mType MarshalType
 	return compressedPayload, payload, nil
 }
 
-func checkSize(payload []byte) bool {
-	if len(payload) >= maxPayloadSize {
+func checkSizeCompressed(payload []byte) bool {
+	if len(payload) >= maxPayloadSizeCompressed {
```

Comment:
Nit. Just return the result of the test directly:`return len(payload) < maxPayloadSizeCompressed`, similar nit for `checkSizeUnCompressed`.